messages/teleporter/message_manager.go

```diff
@@ -323,3 +294,40 @@ func (m *messageManager) parseTeleporterMessage(warpMessageID ids.ID, warpPayloa
 	}
 	return teleporterMessage, nil
 }
+
+func (m *messageManager) getTeleporterMessenger(destinationClient vms.DestinationClient) (*teleportermessenger.TeleporterMessenger, error) {
+	client, ok := destinationClient.Client().(ethclient.Client)
+	if !ok {
+		m.logger.Error(
+			"Destination client is not an Ethereum client.",
+			zap.String("destinationBlockchainID", destinationClient.DestinationBlockchainID().String()),
+		)
+		return nil, errors.New("destination client is not an Ethereum client")
+	}
+
+	// Get the teleporter messenger contract
+	teleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(common.BytesToAddress(m.protocolAddress[:]), client)
```

Comment:
Kind of unrelated to this PR, but does anyone know why `protocolAddress` is type `common.Hash` instead of `common.Address`? @cam-schultz

Reply:
Yeah, this is to support other VMs in the future that may not adhere to EVM 20 bytes addresses.

Reply:
Hmm... Seems like that is just adding complexity for now that won't really help us down the road. In order to support other address lengths, we'd need to rearchitect a lot of the repo. Just about everywhere else uses `common.Address` and our golang bindings are specific to the address length as well.

Reply:
I definitely agree. As evidenced by this [draft PR](https://github.com/ava-labs/awm-relayer/pull/54), accounting for message protocol changes doesn't fit super nicely into the existing architecture. Whenever we go about supporting the next non-EVM and non-Teleporter use case, we should reevaluate the architecture as well. Until then, I think we should not attempt to design interfaces in anticipation of those future use cases. In other words, let's not change the interfaces until we have a second concrete use case to guide that design.

Reply:
Yup agree with yall's points, parts of repo are evm specific, other parts are not. Lets wait for an actual non-evm case to redesign the itnerfaces.

Reply:
> Until then, I think we should not attempt to design interfaces in anticipation of those future use cases. In other words, let's not change the interfaces until we have a second concrete use case to guide that design.

I totally agree with the first sentence here, but the conclusion that leads me to is not to "not change" the interfaces, but to make them no more general than is necessary. 

I think it's kind of dangerous to take a 20-byte address, pass it around as something else, only to convert it back to a 20-byte address later. It's also kind of dangerous to make the desinationClient pass back an interface, if in every case it's using an `ethclient.Client`. I think both of those premature abstractions only lead us to code in strange ways that may be more error-prone as we build around them. I think once the second use-case comes up, that will give the correct indication of how we should go about making the interfaces more general.

Reply:
Agreed, feel free to create an issue to track this. I'd just prefer to not expand the scope of this PR.

Reply:
New issue created by @geoff-vball  thanks https://github.com/ava-labs/awm-relayer/issues/132